File: Data Structures & Algorithms/same-binary-tree/submission-3.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public boolean isSameTree(TreeNode p, TreeNode q) {
        if(p == null & q == null){
            return true;
        }
        boolean result = helper(p, q);
        return result;
    }

    private boolean helper(TreeNode p, TreeNode q){
        if(p == null & q == null){
            return true;
        } else if(p == null || q == null) {
            return false;
        } 
        return  (p.val == q.val) && helper(p.left, q.left) && helper(p.right, q.right);

    }
}

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-3.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        
        level = 0
        res = []
        q = deque()
        if root:
            q.append((root, level))
        while q:
            x, y = q.popleft()
            if y == len(res):
                res.append([])

            res[y].append(x.val)
            
            if x.left:
                q.append((x.left, y+1))
            if x.right:
                q.append((x.right, y+1))
            level += 1

        return res
